async_upnp_client/client_factory.py

```diff
@@ -143,7 +143,15 @@ async def _async_create_service(
         """Retrieve the SCPD for a service and create a UpnpService from it."""
         scpd_url = service_description_el.findtext("device:SCPDURL", None, NS)
         scpd_url = urllib.parse.urljoin(base_url, scpd_url)
-        scpd_el = await self._async_get(scpd_url)
+
+        try:
+            scpd_el = await self._async_get(scpd_url)
+
+        except UpnpXmlParseError as exception:
+            if not self._non_strict:
+                raise UpnpXmlContentError(f"Ignoring bad XML document from URL {scpd_url}") from exception
```

Comment:
In this case you're not ignoring the bad XML document, you're raising a new exception for the caller to deal with.

If you leave it as a bare `raise`, the exception will continue to the caller unchanged, and then the caller can inspect it to determine what went wrong. This all applies only in strict mode, when the caller is probably interested in such things.

Reply:
Many thanks @chishm,
this code was actually suggested to me by @StevenLooman after I committed our first version of the code.
(see the other closed pr: https://github.com/StevenLooman/async_upnp_client/pull/127)

If I'm not wrong, you would change it to:
`raise from exception`

Let's wait for a response by him before pushing a new commit?

Reply:
I think @chishm is right. The `UpnpXmlParseError` contains the actual error, lets use that.

Reply:
Many thanks to both of you for your suggestions and your patience.
I edited this part as:
```
try:
            scpd_el = await self._async_get(scpd_url)

        except UpnpXmlParseError as err:
            if not self._non_strict:
                raise UpnpXmlContentError(err) from err
            _LOGGER.debug("Ignoring bad XML document from URL %s", scpd_url)
            scpd_el = ET.Element(f"{{{NS['service']}}}scpd")
```
Is this part of code ok?
So afterwords I could concentrate just on the tests part.
Thanks again.

Reply:
I think what @chishm said is the way to go, something like this:
```python
        except UpnpXmlParseError as err:
            if not self._non_strict:
                raise
            _LOGGER.debug("Ignoring bad XML document from URL %s", scpd_url)
            scpd_el = ET.Element(f"{{{NS['service']}}}scpd")
```

This will re-raise the `UpnpXmlParseError` exception. This exception (instance) will also include the actual parser error, giving hints to the user of this library. No need to wrap it as another exception.

Reply:
Hi @StevenLooman,
many thanks
pasting your command as-is, gave me
`flake8...................................................................Failed
- hook id: flake8
- exit code: 1

async_upnp_client/client_factory.py:150:9: F841 local variable 'err' is assigned to but never used`

So I edited like this:
` _LOGGER.debug("Ignoring bad XML document from URL %s:\n%s", scpd_url, err)`

hope this 'll be ok.

Reply:
That's a good solution, since the log is often the only place we developers will see what has gone wrong for users.

Another option (for future reference) is to change the `except UpnpXmlParseError as err:` to a plain `except UpnpXmlParseError`.

Reply:
Just one thing, please replace the `\n` with a space. This makes scanning/skimming logs a tiny bit easier.
```python
_LOGGER.debug("Ignoring bad XML document from URL %s: %s", scpd_url, err)
```